### PR TITLE
Navbar and footer 

### PR DESCRIPTION
**Pull request by SophieSjoberg**
## PT link https://www.pivotaltracker.com/story/show/154171885 

Creating a generic footer and navbar with home button and function and cart icon

### CODE REVIEW

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -1,3 +1,20 @@
 Then("I should see {string}") do |content|
   expect(page).to have_content content
 end
+
+Then("{string} is present") do |selector|
+  expect(page).to have_selector :css, "#navigationBar"
+  expect(page).to have_selector :css, "#footer"
+end
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Can't see that you are making use of `selector`

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+
+Then("I would like to see a {string} link") do |link|
+  find_link link
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Hmm... So you `find` the filnk. And then what? 
(if this is supposed to be an assertion, why not use the `expect` keyword?)

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+end
+
+Then("I would like to click on {string} link") do |link|
+  click_link_or_button link
+end
+
+Then("I would like to be redirected to landing page") do
+  visit root_path
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Again, is this an assertion or are you telling Capybara what to do?

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+  expect(page).to have_selector :css, "#navigationBar"
+  expect(page).to have_selector :css, "#footer"
+end
+
+Then("I would like to see a {string} link") do |link|
+  find_link link
+end
+
+Then("I would like to click on {string} link") do |link|
+  click_link_or_button link
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: The title sounds like an assertion but you give a command to Capybarawhat to click on.

File: features/navbar.feature [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Feature: Navbar
+  As a user
+  In order to navigate the site
+  I would like to see a navbar with a home button and a cart button displayed
+
+  Scenario: Display navbar to visitor
+    Given I visit the landing page
+    And "navigationBar" is present
+    And I would like to see a "Cart" link
+    Then I would like to click on "Home" link
```
Comment (tochman) [inline features/navbar.feature]: This wording does not make sense. Please read it out loud and you hear what I mean.

File: features/footer.feature [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Feature: Footer
+  As a user
+  In order to see information about the site
+  I would like to see info about the platform company in the footer
+
+
```
Comment (tochman) [inline features/footer.feature]: Remove extra line please

File: app/assets/stylesheets/application.css [hunk 7]
```diff
@@ -17,3 +17,17 @@
...
+        position: absolute;
+        width: 100%;
+        left: 0;
+        text-align: center;
+        margin:0 auto;
+    }
+}
+.card-footer {
+  margin-top: 100px
+}
```
Comment (tochman) [inline app/assets/stylesheets/application.css]: Please check the formatting in this file. Also, it is much better if you use sass

File: app/views/partials/_footer.html.erb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<div class="card-footer" id="footer">
+  <blockquote class="blockquote text-center mb-0">
+    <p>Kitchen to Couch™</p>
+    <footer class=“blockquote-footer”>Thomas Ochman <cite title=“Source Title”>I don't like carts</cite></footer>
+  </blockquote>
+</div>
```
Comment (tochman) [inline app/views/partials/_footer.html.erb]: Not sure if wrapping the tag `<footer>` in a div and blockquote is good practice. But I love the quote though.

File: app/views/partials/_navbar.html.erb [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    </button>
+    <div class="navbar-collapse collapse" id="collapsingNavbar">
+        <ul class="navbar-nav">
+            <li class="nav-item active">
+                <%= link_to 'Home', root_path, class: 'nav-link' %>
+            </li>
+        </ul>
+        <ul class="navbar-nav ml-auto">
+            <li class="nav-item">
+                <a class="nav-link" href="" data-target="#myModal" data-toggle="modal">Cart</a>
```
Comment (tochman) [inline app/views/partials/_navbar.html.erb]: How about that modal link, what does it do?

File: features/footer.feature [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+  Scenario: Display footer to visitor
+    Given I visit the landing page
+    Then "footer" is present
```
Comment (tochman) [inline features/footer.feature]: is it? or `.. should be present` or `visible`?  Revisit this please.

**Review by tochman — CHANGES_REQUESTED** [review]
Not quite ready for primetime... yet.

File: features/step_definitions/basic_steps.rb [hunk 11]
```diff
@@ -1,3 +1,20 @@
 Then("I should see {string}") do |content|
   expect(page).to have_content content
 end
+
+Then("{string} is present") do |selector|
+  expect(page).to have_selector :css, "#navigationBar"
+  expect(page).to have_selector :css, "#footer"
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: 2 expectations in one step checking for 2 differant (hard coded) selectors? Not sure about this...

File: app/assets/stylesheets/application.css (hunk 7 above)
Comment (NegativeKarma) [reply app/assets/stylesheets/application.css]: Formatting checked, we'll discuss with the others if we want to use sass

File: app/views/partials/_footer.html.erb (hunk 8 above)
Comment (SophieSjoberg) [reply app/views/partials/_footer.html.erb]: It is a standard bootstrap footer template, we put it in now and will style it when we look at front end, is that wrong to do that?

File: app/views/partials/_navbar.html.erb (hunk 9 above)
Comment (SophieSjoberg) [reply app/views/partials/_navbar.html.erb]: Same here with the standard bootstrap navbar template, but I understand what you mean and we'll take it out.

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (SophieSjoberg) [reply features/step_definitions/basic_steps.rb]: ask @diraulo :)

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (SophieSjoberg) [reply features/step_definitions/basic_steps.rb]: The link will be functional later on in the project, is it wrong to just test the presence of it?

File: app/views/partials/_navbar.html.erb (hunk 9 above)
Comment (NegativeKarma) [reply app/views/partials/_navbar.html.erb]: removed the modal, we'll use the link later on when we have the order page

File: features/footer.feature (hunk 6 above)
Comment (NegativeKarma) [reply features/footer.feature]: Done

File: features/footer.feature (hunk 10 above)
Comment (NegativeKarma) [reply features/footer.feature]: Changed to `Then "footer" is visible`

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (NegativeKarma) [reply features/step_definitions/basic_steps.rb]: Changed to `Then I click on "Home" link`

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (NegativeKarma) [reply features/step_definitions/basic_steps.rb]: changed to `Then("I will be redirected to landing page") do`

File: app/views/partials/_footer.html.erb (hunk 8 above)
Comment (tochman) [reply app/views/partials/_footer.html.erb]: Okay, not a blocker for me.

File: app/assets/stylesheets/application.css (hunk 7 above)
Comment (tochman) [reply app/assets/stylesheets/application.css]: I think this kind of decisions (about what preprocessors to use) should be made promptly. That goes for a templating preprocessors as well. Haml ans Sass are the way to go IMO. I have hard time reviewing complex html templates

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (tochman) [reply features/step_definitions/basic_steps.rb]: About what? Is this @diraulo 's code or yours`

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (tochman) [reply features/step_definitions/basic_steps.rb]: This is not a test. This is a command to look for a link and nothing else

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (tochman) [reply features/step_definitions/basic_steps.rb]: I really don't understand... `Then` steps are usually assertion steps. What is going on here?

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (tochman) [reply features/step_definitions/basic_steps.rb]: That does not fix the problem nor  What has `visit root_path` to do with redirect?

File: app/views/partials/_navbar.html.erb [hunk 12]
```diff
@@ -11,7 +11,7 @@
         </ul>
         <ul class="navbar-nav ml-auto">
             <li class="nav-item">
-                <a class="nav-link" href="" data-target="#myModal" data-toggle="modal">Cart</a>
+                <a class="nav-link" href="">Cart</a>
```
Comment (tochman) [inline app/views/partials/_navbar.html.erb]: Please do not use the word Cart. It really sounds so 90-ties

Comment (SophieSjoberg) [reply app/views/partials/_navbar.html.erb]: I love it, I'm a child of the 90's. But maybe your right and I need to refresh my views..

**Pull request merged**